engine/src/main/java/io/zeebe/engine/processing/workflowinstance/CreateWorkflowInstanceProcessor.java

```diff
@@ -51,23 +51,25 @@
   private final MutableElementInstanceState elementInstanceState;
   private final MutableVariableState variablesState;
   private final KeyGenerator keyGenerator;
+  private final StateWriter stateWriter;
 
   public CreateWorkflowInstanceProcessor(
       final WorkflowState workflowState,
       final MutableElementInstanceState elementInstanceState,
       final MutableVariableState variablesState,
-      final KeyGenerator keyGenerator) {
+      final KeyGenerator keyGenerator,
+      final StateWriter stateWriter) {
     this.workflowState = workflowState;
     this.elementInstanceState = elementInstanceState;
     this.variablesState = variablesState;
     this.keyGenerator = keyGenerator;
+    this.stateWriter = stateWriter;
```

Comment:
So here my objection is that I need to pass in the `stateWriter` again, although the `CommandControl` already knows about the state writer. I think this could be improved (see my comment on `CommandControlImpl` for context).